src/plugins/data_views/common/data_views/data_view_lazy.ts

```diff
@@ -0,0 +1,619 @@
+        };
+        return scriptFields;
+      },
+      {}
+    );
+  }
+
+  async getComputedFields({ fieldNames = ['*'] }: { fieldNames: string[] }) {
+    const scriptFields: Record<string, estypes.ScriptField> = {};
+
+    const fieldMap = (
+      await this.getFields({
+        fieldName: fieldNames,
+        fieldTypes: ['date', 'date_nanos'],
+        scripted: false,
+        runtime: false,
+      })
+    ).getFieldMap();
+
+    const docvalueFields = Object.values(fieldMap).map((dateField) => {
+      return {
+        field: dateField.name,
+        format:
+          dateField.esTypes && dateField.esTypes.indexOf('date_nanos') !== -1
+            ? 'strict_date_time'
+            : 'date_time',
+      };
+    });
+
+    each(this.getScriptedFields({ fieldName: fieldNames }), function (field) {
+      scriptFields[field.name] = {
+        script: {
+          source: field.script as string,
+          lang: field.lang,
+        },
+      };
+    });
+
+    return {
+      scriptFields,
+      docvalueFields,
+      runtimeFields: this.getRuntimeMappings(),
+    };
+  }
+
+  getRuntimeMappings(): estypes.MappingRuntimeFields {
+    // @ts-expect-error composite type is not yet supported by es client but it can be forced
+    return this.runtimeFieldMap;
+  }
+
+  /**
+   * Creates static representation of the data view.
+   * @param includeFields Whether or not to include the `fields` list as part of this spec. If not included, the list
+   * will be fetched from Elasticsearch when instantiating a new Data View with this spec.
+   */
+  public async toSpec(params?: { fieldParams?: GetFieldsParams }): Promise<DataViewSpec> {
+    const spec = this.toSpecShared(!!params?.fieldParams);
+
+    if (params?.fieldParams) {
+      const fields: DataViewFieldSpecMap = {};
+      const fieldMap = (await this.getFields(params.fieldParams)).getFieldMap();
+      Object.values(fieldMap).forEach((field) => {
+        fields[field.name] = field.toSpec({
+          getFormatterForField: this.getFormatterForField.bind(this),
+        });
+      });
+      spec.fields = fields;
+    }
+
+    return spec;
+  }
+
+  /**
+   * Creates a minimal static representation of the data view. Fields and popularity scores will be omitted.
+   */
+  public toMinimalSpec(): Omit<DataViewSpec, 'fields'> {
+    // removes `fields`
+    const spec = this.toSpecShared(false);
+
+    if (spec.fieldAttrs) {
+      // removes `count` props (popularity scores) from `fieldAttrs`
+      spec.fieldAttrs = pickBy(
+        mapValues(spec.fieldAttrs, (fieldAttrs) => omit(fieldAttrs, 'count')),
+        (trimmedFieldAttrs) => Object.keys(trimmedFieldAttrs).length > 0
+      );
+
+      if (Object.keys(spec.fieldAttrs).length === 0) {
+        delete spec.fieldAttrs;
+      }
+    }
+
+    return Object.fromEntries(Object.entries(spec).filter(([, v]) => typeof v !== 'undefined'));
+  }
+
+  /**
+   * returns true if dataview contains TSDB fields
+   */
+  async isTSDBMode() {
```

Comment:
I'm guessing there isn't a better way to do this right now, but it seems heavy to fetch all fields just to check for TSDB mode. Is this something we're hoping to improve in the future?

Reply:
I have it noted in https://github.com/elastic/kibana/issues/178594